src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/GitHookTasks.kt

```diff
@@ -33,8 +36,18 @@ open class InstallPrePushHookTask : InstallHookTask("pre-push") {
  * Install or update a kotlinter-gradle hook.
  */
 abstract class InstallHookTask(@get:Internal val hookFileName: String) : DefaultTask() {
+
     @Input
-    val gitDirPath: Property<String> = project.objects.property(default = ".git")
+    val gitDirPath = project.objects.property(default = ".git")
+
+    @get:InputDirectory
+    internal val projectDir: DirectoryProperty = project.objects.directoryProperty().apply {
+        set(project.layout.projectDirectory)
+    }
+
+    private val gitRepo: Repository =
+        RepositoryBuilder().findGitDir(projectDir.get().asFile)
```

Comment:
❔ 
What's the purpose of `projectDir` property? What would happen if you instead wrote:
```suggestion
    private val gitRepo = RepositoryBuilder().findGitDir(project.projectDir)
```
?